src/pages/benchmarks/[benchmark_id]/index.tsx

```diff
@@ -95,20 +115,35 @@ const BenchmarkPage = () => {
     }
   };
 
-  // BUG: Updating counters before the timer goes off results in changes being reverted.
-  const onNoteChange = (e: React.FormEvent<HTMLInputElement>) => {
-    setNotesValue((e.target as HTMLInputElement).value);
+  const onNoteChange = (e: React.FormEvent<HTMLTextAreaElement>) => {
+    setNotesInputValue((e.target as HTMLInputElement).value);
   };
+
+  //Only updates db after inputs have stopped for 1 sec
   useDebounce(
     () => {
-      //Only update db after typing has stopped for 1 sec
-      handleUpdate({
-        notes: notesValue,
-      });
+      if (hasInputChanged) {
+        handleUpdate({
+          notes: notesInputValue,
+          success: successInputValue,
+          unsuccess: unsuccessInputValue,
+        });
+      }
     },
     1000,
-    [notesValue]
+    [notesInputValue, successInputValue, unsuccessInputValue]
   );
+
+  // BUG?: Sometimes if the user reloads/navigates away and confirms, the update has time to go through and data is saved. Is this something we should fix?
+  const checkUnsavedData = useCallback(() => {
+    if (hasInputChanged) {
+      return true;
+    }
+    return false;
```

Comment:
```suggestion
    return hasInputChanged
```

Reply:
LOL 🤦‍♂️

Reply:
Actually I'll do you one better. We don't even need this callback since we can just pass hasInputChanged in the useConfirmBeforeLeave hook.